src/python/pants/help/help_info_extracter.py

```diff
@@ -246,10 +248,22 @@ class RuleInfo:
 class AllHelpInfo:
     """All available help info."""
 
-    scope_to_help_info: dict[str, OptionScopeHelpInfo]
-    name_to_goal_info: dict[str, GoalHelpInfo]
-    name_to_target_type_info: dict[str, TargetTypeHelpInfo]
-    rule_output_type_to_rule_infos: dict[str, tuple[RuleInfo, ...]]
+    scope_to_help_info: LazyFrozenDict[str, OptionScopeHelpInfo]
+    name_to_goal_info: LazyFrozenDict[str, GoalHelpInfo]
+    name_to_target_type_info: LazyFrozenDict[str, TargetTypeHelpInfo]
+    rule_output_type_to_rule_infos: LazyFrozenDict[str, tuple[RuleInfo, ...]]
+
+    def asdict(self) -> dict[str, Any]:
+        return {
+            field: {
+                # This will be simplified when the rule infos are flattened...
```

Comment:
Let's leave this out of the code imo. It's a useful exercise to assume followup work will _not_ happen - would the comment still make sense? No, because "this will" wouldn't be true.

Instead, either a comment on the PR itself is helpful for stuff like this, or rewriting it to be a TODO comment.